src/v/archival/ntp_archiver_service.h

```diff
@@ -359,14 +371,15 @@ class ntp_archiver {
     /// in the expectation that we will be called again in the main upload loop.
     /// Returns true if something was uploaded (_projected_manifest_clean_at
     /// will have been updated if so)
-    ss::future<bool> maybe_upload_manifest();
+    ss::future<bool> maybe_upload_manifest(std::string_view upload_ctx);
```

Comment:
i wonder if we should take `const char*`  here instead of a std::string_view (and more generally do this in the tree when the intention is a string that has static lifetime). the reason is that it's pretty easy to now accidentally pass in something that doesn't have static lifetime (which i think is assumed here--all the contexts are string literals), such as `maybe(fmt::format(...))` which would likely segfault if maybe_upload_manifest suspended before using it and the caller didn't co_await. const char*.

Reply:
started a discussion in team-core

Reply:
also this is a nit :)

Reply:
It's a good point. I was aware of the foot-gun but thought it's okay since I changed all the callers.
I really wish c++ had a decent sum type or at least some sort of decent static enum of strings. I updated the code.